pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go

```diff
@@ -1041,14 +1041,8 @@ var _ = Describe("OperatingSystemConfig", func() {
 
 		It("is different for different worker.cri configurations", func() {
 			containerDKey := Key(workerName, semver.MustParse("1.2.3"), &gardencorev1beta1.CRI{Name: gardencorev1beta1.CRINameContainerD})
-			dockerKey := Key(workerName, semver.MustParse("1.2.3"), &gardencorev1beta1.CRI{Name: gardencorev1beta1.CRINameDocker})
-			Expect(containerDKey).NotTo(Equal(dockerKey))
-		})
-
-		It("is the same for `cri=nil` and `cri.name=docker`", func() {
```

Comment:
Shouldn't we keep this test, but use `containerd` instead of `docker`?

Reply:
I think this won't work since we have modified 
```
if criConfig != nil && criConfig.Name != gardencorev1beta1.CRINameDocker
```
and doesn't rely on defaults. The cri name is defaulted in `defaults.go` but the controller doesn't do it.

Reply:
Right, thanks for the explanation.